modules/components/src/AdvancedFacetView/FacetView.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import FacetViewNode from './FacetViewNode';
+import $ from 'jquery';
+import { debounce } from 'lodash';
+
+const serializeToDomId = path => path.split('.').join('__');
+const serializeDomIdToPath = path => path.split('.').join('__');
+
+export default class FacetView extends React.Component {
+  state = {
+    isAnimating: false,
+  };
+  componentWillReceiveProps({
+    selectedMapping,
+    path: selectedPath,
+    aggregations,
+    disPlayTreeData,
+  }) {
+    if (selectedPath) {
+      this.scrollToPath(selectedPath);
+    }
+  }
+
+  scrollToPath = path => {
+    const targetElementId = serializeToDomId(path);
+    const targetElement = $(this.root).find(`#${targetElementId}`);
+    if (targetElement) {
+      this.setState({ isAnimating: true });
+      $(this.root)
+        .stop()
+        .animate(
+          {
+            scrollTop:
+              $(this.root).scrollTop() +
+              targetElement.offset().top -
+              $(this.root).offset().top,
+          },
+          {
+            duration: 500,
+            complete: () => this.stopScrollAnimation(),
+          },
+        );
+    }
+  };
+
+  stopScrollAnimation = () => {
+    $(this.root).stop();
+    this.setState({ isAnimating: false });
+  };
+
+  onScroll = debounce(e => {
```

Comment:
What do you think of using [react-waypoint ](https://github.com/brigade/react-waypoint) instead of listening on scroll? might simplify it a bit.

Reply:
I will opt for the refactoring we talked about yesterday to remove the extra feature so I don't we'll need to worry about this :)